pebbles/views/export_stats.py

```diff
@@ -0,0 +1,372 @@
+from flask.ext.restful import reqparse
+from flask import Blueprint as FlaskBlueprint
+from datetime import datetime as dt
+from operator import itemgetter
+
+from pebbles.server import restful, app
+from pebbles.views.commons import auth
+from pebbles.utils import requires_admin
+from pebbles.rules import (apply_rules_export_institutions, apply_rules_export_monthly_instances,
+                           apply_rules_export_quartals)
+
+export_stats = FlaskBlueprint('export_stats', __name__)
+
+date_format = '%Y-%m-%d'
+
+month_count = 12
+
+# institution types
+institution_list = app.config['HAKA_INSTITUTION_LIST']
+
+# Quartal definition
+Q1 = [1, 2, 3]
+Q2 = [4, 5, 6]
+Q3 = [7, 8, 9]
+Q4 = [10, 11, 12]
+
+
+class ExportInstitutions(restful.Resource):
+
+    parser = reqparse.RequestParser()
+    parser.add_argument('start', type=str)
+    parser.add_argument('end', type=str)
+    parser.add_argument('filter', type=str)
+    parser.add_argument('exclude', type=bool)
+
+    @auth.login_required
+    @requires_admin
+    def get(self):
+        args = self.parser.parse_args()
+        if args.get('start') is None and args.get('end') is None:
+            query = apply_rules_export_institutions()
+        else:
+            query = apply_rules_export_institutions(args)
+        institutions = query.all()
+        results = {}
+        for inst in institutions:
+            email = str(inst._email)
+            domain = email.split("@")[1]
+
+            if domain in results:
+                results[domain] += 1
+            else:
+                results[domain] = 1
+        results = sorted(results.items(), key=itemgetter(1), reverse=True)
+
+        filter = args.get('filter')
```

Comment:
Invalid domain name also exports data. Validate.

Reply:
Also the validation for date and domain are same so put them in a single function and call them whenever needed.